### ISSUE TITLE
dispcal args are getting unecessarily quoted preventing it to run properly

### ISSUE THREAD

**Issue by eoyilmaz** [eoyilmaz/displaycal-py3#410]
@eoyilmaz Hmmm I met another bug, not sure if it is related so I post it here, feel free to move to a new issue:

```
--------------------------------------------------------------------------------
Calibrate & profile
DisplayCAL: luminance: 120.0
cmd: D:\Tools\Argyll_V3.3.0\bin\dispcal.exe
args: ['-v2', '-d1', '-c1', '-yl', '-P0.750780437044745,0.42269326683291775,1.4941451990632322', '-H', '-Ibw', '-qv', '-w0.3127,0.329', '-b120.0', '-gs', '-f1.0', '-A4.0', 'C:\\Users\\wiagn\\AppData\\Local\\Temp\\DisplayCAL-x95qfljt\\ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX']
--------------------------------------------------------------------------------
Session log: ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX

Working directory:
  C:\
   Users\
    wiagn\
     AppData\
      Local\
       Temp\
        DisplayCAL-x95qfljt\

Command line:
  D:\Tools\Argyll_V3.3.0\bin\dispcal.exe
    -v2
    -d1
    -c1
    -yl
    "-P0.750780437044745,0.42269326683291775,1.4941451990632322"
    -H
    -Ibw
    -qv
    "-w0.3127,0.329"
    -b120.0
    -gs
    -f1.0
    -A4.0
    "ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX"

DisplayCAL: Starting interaction with subprocess
Setting up the instrument
Instrument Type:   ColorMunki
Serial Number:     4103263
Firmware version:  512
Chip ID:           01-F03AE41E00008F
Version string:    'i1Studio FW V2.00  Build Nr. 1310'
Calibration Ver.:  6
Production No.:    215981
Display type ignored - instrument doesn't support display type selection
Set instrument sensor to calibration position,
 and then hit any key to continue,
DisplayCAL: Detected instrument calibration message
DisplayCAL: Prompting to calibrate instrument
DisplayCAL: Waiting for send buffer
DisplayCAL: About to calibrate instrument
DisplayCAL: Send buffer received:
DisplayCAL: Sending buffer:
 or hit Esc or Q to abort:
Calibration complete
DisplayCAL: Detected instrument calibration complete message
Place instrument on test window.
DisplayCAL: Waiting for send buffer
DisplayCAL: Detected instrument placement (screen/spot) message
DisplayCAL: Prompting to place instrument on screen
DisplayCAL: Instrument on screen
DisplayCAL: Send buffer received:
DisplayCAL: Sending buffer:
Hit Esc or Q to give up, any other key to continue:
Display type is 'l'
Target white = xy 0.312700 0.329000
Target white brightness = 120.000000 cd/m^2
Target black brightness = native brightness
Target gamma = sRGB curve
Display adjustment menu:
Press 1 .. 7
1) Black level (CRT: Offset/Brightness)
2) White point (Color temperature, R,G,B, Gain/Contrast)
3) White level (CRT: Gain/Contrast, LCD: Brightness/Backlight)
4) Black point (R,G,B, Offset/Brightness)
5) Check all
6) Measure and set ambient for viewing condition adjustment
7) Continue on to calibration
8) Exit
DisplayCAL: Waiting for send buffer
DisplayCAL: Send buffer received: 7
DisplayCAL: Sending buffer: 7
Commencing display calibration
Measured display update delay of 0 msec, using delay of 100 msec & 2 msec inst reaction
Patch 9 of 9
Black = XYZ 0.1201 0.1224 0.2559
Red   = XYZ 58.809 26.186  0.371
Green = XYZ 34.106 85.848  9.558
Blue  = XYZ 17.565  7.703 94.082
White = XYZ 115.721 121.773 134.272
Automatic black point hue correction level = 0.40
Patch 41 of 40
Target white value is XYZ 114.054711 120.000000 130.686930 [xy 0.312700 0.329000]
Adjusted target black XYZ 0.1186 0.1224 0.2074, Lab 0.921 0.077 -0.883
Target black after min adjust: XYZ 0.1463 0.1510 0.2559, Lab 1.137 0.096 -1.090
Gamma curve input offset = 0.000000, output offset = 0.001258, power = 0.000000
Total Iteration 1, Final Samples = 16 Final Repeat threshold = 1.500000
Creating initial calibration curves...
Doing iteration 1/1 with 16 sample points and repeat threshold of 1.500000 DE
Patch 1 of 16 DE 0.143204, W.DE 0.143204, W.peqDE 0.143204, OK ( < 1.500000)
Patch 2 of 16 DE 0.873701, W.DE 0.873701, W.peqDE 0.838967, OK ( < 1.500000)
Patch 3 of 16 DE 1.842919, W.DE 1.842919, peqDE 1.840868, Repeat
Patch 3 of 16 DE 0.623360, W.DE 0.623360, W.peqDE 0.826351, OK ( < 1.500000)
Patch 4 of 16 DE 0.498970, W.DE 0.498970, W.peqDE 0.955230, OK ( < 1.500000)
Patch 5 of 16 DE 0.266657, W.DE 0.266657, W.peqDE 0.676907, OK ( < 1.500000)
Patch 6 of 16 DE 0.954280, W.DE 0.954280, W.peqDE 0.705705, OK ( < 1.500000)
Patch 7 of 16 DE 0.764217, W.DE 0.764217, W.peqDE 0.497129, OK ( < 1.500000)
Patch 8 of 16 DE 1.120045, W.DE 1.120045, W.peqDE 0.824856, OK ( < 1.500000)
Patch 9 of 16 DE 1.574833, W.DE 1.574833, peqDE 1.379406, Repeat
Patch 9 of 16 DE 0.360873, W.DE 0.360873, W.peqDE 0.743965, OK ( < 1.500000)
Patch 10 of 16 DE 1.338679, W.DE 1.338679, W.peqDE 1.293744, OK ( < 1.500000)
Patch 11 of 16 DE 0.838031, W.DE 0.838031, W.peqDE 0.327562, OK ( < 1.500000)
Patch 12 of 16 DE 1.488476, W.DE 1.488476, W.peqDE 1.298058, OK ( < 1.500000)
Patch 13 of 16 DE 2.189610, W.DE 2.189610, peqDE 2.419363, Repeat
Patch 13 of 16 DE 0.422971, W.DE 0.422971, W.peqDE 0.706800, OK ( < 1.500000)
Patch 14 of 16 DE 1.877162, W.DE 1.877162, peqDE 1.964881, Repeat
Patch 14 of 16 DE 0.504794, W.DE 0.504794, W.peqDE 0.346588, OK ( < 0.931957)
Patch 15 of 16 DE 1.021826, W.DE 1.021826, peqDE 1.302922, Repeat
Patch 15 of 16 DE 0.714616, W.DE 0.714616, peqDE 1.019961, Repeat
Patch 15 of 16 DE 0.541251, W.DE 0.541251, peqDE 0.159096, Repeat
Patch 15 of 16 DE 0.541251, W.DE 0.541251, peqDE 0.159096, Repeat
Patch 15 of 16 DE 0.328776, W.DE 0.328776, W.peqDE 0.256575, OK ( < 0.458969)
Patch 16 of 16 DE 0.557437, W.DE 0.557437, peqDE 0.850528, Repeat
Patch 16 of 16 DE 0.536246, W.DE 0.536246, peqDE 0.834563, Repeat
Patch 16 of 16 DE 0.428402, W.DE 0.428402, peqDE 0.765459, Repeat
Patch 16 of 16 DE 0.371658, W.DE 0.371658, peqDE 0.724274, Repeat
Patch 16 of 16 DE 0.220898, W.DE 5.042895, peqDE 0.724274, Repeat (got worse)
Patch 16 of 16 DE 0.257589, W.DE 0.257589, peqDE 0.584616, Repeat
Patch 16 of 16 DE 0.187503, W.DE 37.327563, peqDE 0.584616, Repeat (got worse)
Patch 16 of 16 DE 0.158906, W.DE 19.673131, peqDE 19.676647, Repeat
Patch 16 of 16 DE 0.134825, W.DE 53.598848, peqDE 0.584616, Repeat (got worse)
Patch 16 of 16 DE 0.217541, W.DE 0.217541, W.peqDE 0.547720, Fail ( > 0.458969)
Brightness error = 0.221193 cd/m^2 (is 120.221193, should be 120.000000)
White point error = 0.124156 deltaE
Maximum neutral error (@ 0.120656) = 1.488476 deltaE
Average neutral error = 0.671536 deltaE
Failed to meet target 1.500000 delta E, got worst case 0.217541
Number of measurements taken = 39
Computing update to calibration curves...
Black drift was 0.078382 DE
White drift was 0.151113 DE
The instrument can be removed from the screen.
Written calibration file 'ULTRAGEAR.cal'
DisplayCAL: Reached EOF (OK)
dispcal exitcode: 0
result: True
dst_pathname: D:\ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX\ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX
cal         : C:\Users\wiagn\AppData\Local\Temp\DisplayCAL-x95qfljt\ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX.cal
result      : No calibration file has been created.
Moving files of incomplete run to C:\Users\wiagn\AppData\Roaming\DisplayCAL\incomplete\ULTRAGEAR #1 2024-10-12 19-51 120cdm² 0.3127x 0.329y sRGB VF-S XYZLUT+MTX
┌──────────────────────────────────────────────────────────────────────────────┐
│ No calibration file has been created.                                        │
└──────────────────────────────────────────────────────────────────────────────┘
```

_Originally posted by @Headcrabed in https://github.com/eoyilmaz/displaycal-py3/issues/406#issuecomment-2408538196_

**Comment by eoyilmaz** [comment]
@headcrabed can you test this branch please:

```shell
cd %HOME%
git clone https://github.com/eoyilmaz/displaycal-py3.git
cd displaycal-py3
git checkout 410-dispcal-args-are-getting-unecessarily-quoted-preventing-it-to-run-properly
C:\Program Files\Python311\Scripts\pip install -r requirements.txt -r requirements-dev.txt
C:\Program Files\Python311\python -m build
C:\Program Files\Python311\Scripts\pip install --force-reinstall dist\DisplayCAL-3.9.*.whl
```

**Comment by Headcrabed** [comment]
@eoyilmaz No, this branch does not help on this issue

```
--------------------------------------------------------------------------------
校准 & 配置文件
DisplayCAL: luminance: 120.0
cmd: D:\Tools\Argyll_V3.3.0\bin\dispcal.exe
args: ['-v2', '-d1', '-c1', '-yl', '-P0.7013527575442248,0.41770573566084784,1.4941451990632322', '-H', '-Ibw', '-qv', '-w0.3127,0.329', '-b120.0', '-gs', '-f1.0', '-A4.0', 'C:\\Users\\wiagn\\AppData\\Local\\Temp\\DisplayCAL-_d3ganud\\ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX']
--------------------------------------------------------------------------------
Session log: ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX

工作目录：
  C:\
   Users\
    wiagn\
     AppData\
      Local\
       Temp\
        DisplayCAL-_d3ganud\

命令行：
  D:\Tools\Argyll_V3.3.0\bin\dispcal.exe
    -v2
    -d1
    -c1
    -yl
    "-P0.7013527575442248,0.41770573566084784,1.4941451990632322"
    -H
    -Ibw
    -qv
    "-w0.3127,0.329"
    -b120.0
    -gs
    -f1.0
    -A4.0
    "ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX"

DisplayCAL: Starting interaction with subprocess
Setting up the instrument
Instrument Type:   ColorMunki
Serial Number:     4103263
Firmware version:  512
Chip ID:           01-F03AE41E00008F
Version string:    'i1Studio FW V2.00  Build Nr. 1310'
Calibration Ver.:  6
Production No.:    215981
Display type ignored - instrument doesn't support display type selection
Set instrument sensor to calibration position,
 and then hit any key to continue,
DisplayCAL: Detected instrument calibration message
DisplayCAL: Prompting to calibrate instrument
DisplayCAL: Waiting for send buffer
DisplayCAL: About to calibrate instrument
DisplayCAL: Send buffer received:
DisplayCAL: Sending buffer:
 or hit Esc or Q to abort:
Calibration complete
DisplayCAL: Detected instrument calibration complete message
Place instrument on test window.
DisplayCAL: Detected instrument placement (screen/spot) message
DisplayCAL: Prompting to place instrument on screen
DisplayCAL: Waiting for send buffer
DisplayCAL: Instrument on screen
DisplayCAL: Send buffer received:
DisplayCAL: Sending buffer:
Hit Esc or Q to give up, any other key to continue:
Display type is 'l'
Target white = xy 0.312700 0.329000
Target white brightness = 120.000000 cd/m^2
Target black brightness = native brightness
Target gamma = sRGB curve
Display adjustment menu:
Press 1 .. 7
1) Black level (CRT: Offset/Brightness)
2) White point (Color temperature, R,G,B, Gain/Contrast)
3) White level (CRT: Gain/Contrast, LCD: Brightness/Backlight)
4) Black point (R,G,B, Offset/Brightness)
5) Check all
6) Measure and set ambient for viewing condition adjustment
7) Continue on to calibration
8) Exit
DisplayCAL: Waiting for send buffer
DisplayCAL: Send buffer received: 7
DisplayCAL: Sending buffer: 7
Commencing display calibration
Measured display update delay of 0 msec, using delay of 100 msec & 2 msec inst reaction
Patch 9 of 9
Black = XYZ 0.1290 0.1313 0.2685
Red   = XYZ 59.298 26.413  0.369
Green = XYZ 34.334 86.478  9.630
Blue  = XYZ 23.017  9.862 124.625
White = XYZ 116.584 122.753 134.872
Automatic black point hue correction level = 0.37
Patch 41 of 40
Target white value is XYZ 114.054711 120.000000 130.686930 [xy 0.312700 0.329000]
Adjusted target black XYZ 0.1274 0.1313 0.2221, Lab 0.988 0.091 -0.943
Target black after min adjust: XYZ 0.1540 0.1587 0.2685, Lab 1.195 0.110 -1.140
Gamma curve input offset = 0.000000, output offset = 0.001322, power = 0.000000
Total Iteration 1, Final Samples = 16 Final Repeat threshold = 1.500000
Creating initial calibration curves...
Doing iteration 1/1 with 16 sample points and repeat threshold of 1.500000 DE
Patch 1 of 16 DE 0.327701, W.DE 0.327701, W.peqDE 0.327701, OK ( < 1.500000)
Patch 2 of 16 DE 1.118608, W.DE 1.118608, W.peqDE 1.011532, OK ( < 1.500000)
Patch 3 of 16 DE 1.528069, W.DE 1.528069, peqDE 1.853896, Repeat
Patch 3 of 16 DE 0.553531, W.DE 0.553531, W.peqDE 0.270000, OK ( < 1.500000)
Patch 4 of 16 DE 0.578614, W.DE 0.578614, W.peqDE 0.952163, OK ( < 1.500000)
Patch 5 of 16 DE 0.314400, W.DE 0.314400, W.peqDE 0.576511, OK ( < 1.500000)
Patch 6 of 16 DE 0.231484, W.DE 0.231484, W.peqDE 0.452936, OK ( < 1.500000)
Patch 7 of 16 DE 0.970629, W.DE 0.970629, W.peqDE 0.822098, OK ( < 1.500000)
Patch 8 of 16 DE 1.378466, W.DE 1.378466, peqDE 1.583460, Repeat
Patch 8 of 16 DE 0.385537, W.DE 0.385537, W.peqDE 0.796984, OK ( < 1.500000)
Patch 9 of 16 DE 2.037738, W.DE 2.037738, peqDE 2.043521, Repeat
Patch 9 of 16 DE 0.382118, W.DE 0.382118, W.peqDE 0.131308, OK ( < 1.500000)
Patch 10 of 16 DE 1.580579, W.DE 1.580579, peqDE 1.700914, Repeat
Patch 10 of 16 DE 0.695878, W.DE 0.695878, W.peqDE 0.427112, OK ( < 1.500000)
Patch 11 of 16 DE 0.953592, W.DE 0.953592, W.peqDE 0.644913, OK ( < 1.500000)
Patch 12 of 16 DE 1.621488, W.DE 1.621488, peqDE 1.635348, Repeat
Patch 12 of 16 DE 0.194473, W.DE 0.194473, W.peqDE 0.673618, OK ( < 1.500000)
Patch 13 of 16 DE 2.227804, W.DE 2.227804, peqDE 2.196677, Repeat
Patch 13 of 16 DE 0.503906, W.DE 0.503906, W.peqDE 0.298426, OK ( < 1.500000)
Patch 14 of 16 DE 1.928123, W.DE 1.928123, peqDE 2.016994, Repeat
Patch 14 of 16 DE 0.543306, W.DE 0.543306, W.peqDE 0.918279, OK ( < 0.932041)
Patch 15 of 16 DE 1.087683, W.DE 1.087683, peqDE 1.137493, Repeat
Patch 15 of 16 DE 0.688914, W.DE 0.688914, peqDE 0.731652, Repeat
Patch 15 of 16 DE 0.265878, W.DE 0.265878, peqDE 0.476272, Repeat
Patch 15 of 16 DE 0.265878, W.DE 0.265878, peqDE 0.476272, Repeat
Patch 15 of 16 DE 0.364652, W.DE 0.364652, peqDE 0.535884, Repeat
Patch 15 of 16 DE 0.242655, W.DE 9.868287, peqDE 9.867091, Repeat
Patch 15 of 16 DE 0.175768, W.DE 0.175768, W.peqDE 0.303284, OK ( < 0.459019)
Patch 16 of 16 DE 0.635909, W.DE 0.635909, peqDE 0.615516, Repeat
Patch 16 of 16 DE 0.552941, W.DE 0.552941, peqDE 0.533858, Repeat
Patch 16 of 16 DE 0.534651, W.DE 0.534651, peqDE 0.502443, Repeat
Patch 16 of 16 DE 0.351677, W.DE 0.351677, W.peqDE 0.322608, OK ( < 0.459019)
Brightness error = 0.309691 cd/m^2 (is 120.309691, should be 120.000000)
White point error = 0.311897 deltaE
Maximum neutral error (@ 0.895486) = 1.118608 deltaE
Average neutral error = 0.517576 deltaE
Number of measurements taken = 38
Computing update to calibration curves...
Black drift was 0.036571 DE
White drift was 0.265222 DE
The instrument can be removed from the screen.
Written calibration file 'ULTRAGEAR.cal'
DisplayCAL: Reached EOF (OK)
dispcal exitcode: 0
result: True
dst_pathname: D:\ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX\ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX
cal         : C:\Users\wiagn\AppData\Local\Temp\DisplayCAL-_d3ganud\ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX.cal
result      : 校准文件未创建。
Moving files of incomplete run to C:\Users\wiagn\AppData\Roaming\DisplayCAL\incomplete\ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX
┌──────────────────────────────────────────────────────────────────────────────┐
│ 校准文件未创建。                                                                     │
└──────────────────────────────────────────────────────────────────────────────┘
```

**Comment by eoyilmaz** [comment]
That is weird I still see:

```shell
D:\Tools\Argyll_V3.3.0\bin\dispcal.exe
    -v2
    -d1
    -c1
    -yl
    "-P0.7013527575442248,0.41770573566084784,1.4941451990632322"
    -H
    -Ibw
    -qv
    "-w0.3127,0.329"
    -b120.0
    -gs
    -f1.0
    -A4.0
    "ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX"
```

The `"-P0.7013527575442248,0.41770573566084784,1.4941451990632322"` and `"-w0.3127,0.329"` shouldn't be there. Let me create a proper test.

**Comment by eoyilmaz** [comment]
That's not the problem as I supposed it was.

**Comment by eoyilmaz** [comment]
@Headcrabed can you try using a shorter name which also doesn't have spaces in it.

**Comment by eoyilmaz** [comment]
The problem is, I believe that the calibration file is written as `'ULTRAGEAR.cal'` where as it should have been `ULTRAGEAR #1 2024-10-13 00-06 120cdm² 0.3127x 0.329y sRGB VF-S 1xCurve+MTX`.

**Comment by eoyilmaz** [comment]
can you please confirm if it is working properly when you update the profile name to something that doesn't use any spaces.

**Comment by Headcrabed** [comment]
@eoyilmaz Sorry for late reply, I was busy at that time. After removing all spaces in output filename, it works!

**Comment by Headcrabed** [comment]
@eoyilmaz Btw I was using the previous 410 branch without the commit 1 hour ago for testing. Does the new commit need to be tested?

**Comment by eoyilmaz** [comment]
no it's okay, 410 doesn't anything that makes a different over the develop branch. Thank you, and closing this. But, definitelly we need to solve profile paths with spaces under Windows... in the future :)

**Comment by Headcrabed** [comment]
Besides I think we can change the default naming rule for freshly installed DisplayCAL? This would reduce the possibility that new users are blocked by this issue.

**Comment by Headcrabed** [comment]
@eoyilmaz Seems chromarity coordinates is still having space in generated file name:
![图片](https://github.com/user-attachments/assets/ee189501-4154-4742-a894-4c5c6232dc4b)

**Comment by eoyilmaz** [comment]
> Besides I think we can change the default naming rule for freshly installed DisplayCAL? This would reduce the possibility that new users are blocked by this issue.

This is done with ea06bc5

> @eoyilmaz Seems chromarity coordinates is still having space in generated file name:

yes, I think we should update the code so that it works with spaces as it was before.

**Comment by Headcrabed** [comment]
@eoyilmaz Have you fixed the issue that there can't be any spaces in filename?

**Comment by eoyilmaz** [comment]
Over #424, I've fixed the original issue and working on that issue now
